packages/webcomponents/src/components/payment-details/payment-details.tsx

```diff
@@ -30,111 +29,49 @@ import { config } from '../../../config';
 */
 @Component({
   tag: 'justifi-payment-details',
-  styleUrl: 'payment-details.scss',
   shadow: true,
 })
 
 export class PaymentDetails {
   @Prop() paymentId: string;
   @Prop() authToken: string;
-  @State() payment: Payment;
-  @State() loading: boolean = true;
-  @State() errorMessage: string;
 
-  @Watch('paymentId')
-  @Watch('authToken')
-  updateOnPropChange() {
-    this.fetchData();
+  @State() getPaymentDetails: Function;
```

Comment:
I'm not sure how `@State` works with a function, but I do know that deep comparisons don't usually take place when Stencil is deciding to do a re-render. I think since `initializeGetPaymentDetails` creates a _new_ function using `makeGetPaymentDetails` it should cause a re-render. You may want to double check if changing `paymentId` or `authToken` causes a re-render and for `payment-details-core` to be re-initialized just to be sure

Reply:
Thanks for pointing that out! You're right about `@State()` and function references. In our case, since `initializeGetPaymentDetails` assigns a new function, it does trigger a re-render, ensuring `payment-details-core` updates as expected with `paymentId` or `authToken` changes. This aligns with our real-world usage where the component behaves as intended. I'll double-check to confirm the behavior remains consistent. Appreciate the insight!